### DISCUSSION TITLE
It is possible to configure fluxCD monitor and pull manifest files changes to from AWS S3 instead of Git repository

### DISCUSSION THREAD

**Discussion by baoquach14** [fluxcd/flux2#2249] [Q&A] [answered]
Hi All

We would like to deploy new or patch version our Application Docker containers to our customer production sites where Kubernetes cluster are install on Premise as Virtual servers. 

I am thinking of using FluxCD for deployment management of our Docker containers to Kubernetes clusters to our customer production environment around the world.

It is possible to configure fluxCD to monitor and pull manifest files changes to from AWS S3 instead of Git repository. It would trigger a "pipeline" with a manual approval step, where a human operator could manually approve/reject. Once an approval is received, the pipeline would do its action (deploying the new manifest (deployment descriptor) yaml file and new Docker container images to Kubernetes Cluster.

Thank you for your help and support.

Regards, Bao

**Comment by stefanprodan** [comment, marked as answer]
Yes, Flux supports pulling manifests from AWS S3 and any other S3 compatible storage like Minio, Google Cloud storage, etc

Please see the docs here: https://fluxcd.io/docs/components/source/buckets/#spec-examples

**Comment by baoquach14** [comment]
Hi Stefanprodan

Thankyou very much for answering my questions.

Here my typical deployment use case for deploying a Microservice from Kubernetes Master node on Premise to Edge device (KubeEdge or MicroK8) of our costumers site.
1. the following files are locate in AWS S3 
   - s3://au-nsw-s3-100000032685224/prod/spat/spat-mod-v1.0.1-edge1.yaml ( contains site manifest file deploy and run SPAT module Docker container. Note SPAT module Docker container is location in AWS ECR
   -   s3://au-nsw-s3-100000032685224/prod/spat/SPAT.xml
   -   s3://au-nsw-s3-100000032685224/prod/spat/MAP_4443.uper
   -  The MAP_4443.uper and SPAT.xml files is use by SPAT module Docker container. These files need to be deploy on Edge device in physicals location of /opt/spat/config before SPAT module Docker container. 
 
Base on https://fluxcd.io/docs/components/source/buckets/#spec-examples you provided.
I assume that, I have to create an Artifact as  gzip compressed tar file , for example spat-mod-v1.0.1-edge1.gz.tar. which included spat-mod-v1.0.1-edge1.yaml, SPAT.xml and MAP_4443.uper files. Then, I need to create an Artifact manifest file and upload to AWS S3 bucket where Flux will monitor the change. 

Can you please explain how fluxcd will deploy the SPAT module Docker container and dependencies files to Edge device.
Can you configure Flux to save its deployment workflow in Database instead of Git version control. 

Thankyou for your help and support.

Regards, Bao

**Reply by stefanprodan** [reply]
Flux has no support for dealing with manifests other than Kubernetes YAMLs. You need to write your own controller, see here an example https://fluxcd.io/docs/gitops-toolkit/source-watcher/

**Reply by baoquach14** [reply]
Thank you Stefanprodan for suggestion 
can you please let me if possible to configure Flux to save its deployment workflow in Database instead of Git version control.

Regards, Bao
